Add documentation for the web administration gui. (#1592)
Mailu/Mailu: docs/webadministration.rst

For brevity the web administration interface will now be mentioned as admin gui.
It offers the following configuration options:

* change display name.

* change the logged in user's password.

* change user defined spam filter tolerance.

* configure automatic forwarding.

* configure automatic email replies (out of office replies).

* configure fetchmail for automatic email retrieval (from 3rd-party servers).

* configure application passwords.

* send broadcast messages to all users.

* configure global administration users.

* configure relayed domains.

* access Rspamd webui.

* Configure all email domains served by Mailu, including:

    * generating dkim and dmarc keys for a domain.
    
    * view email domain information on how to configure your SPF, DMARC, DKIM and MX dns records for an email domain.
    
    * Add new email domains.
    
    * For existing domains, configure users, quotas, aliases, administrators and alternative domain names.
    
* access the webmail site.

* lookup settings for configuring your email client.

Access the web administration interface
---------------------------------------

> The admin GUI is accessed via the URL `https://<my domain>/admin` for default deployments.

Reviewer: ```suggestion
The admin GUI is by default accessed via the URL `https://<my domain>/admin`, when it's enabled via setup or by manually setting `ADMIN=true` in `mailu.env`.
```

At least using the setup utility it's not default.

The user defined spam filter tolerance determines when an email is classified as ham (moved to the inbox folder) or spam (moved to the junk folder). 
The default value is 80%. The lower the spam filter tolerance, the more false positives (ham classified as spam). The higher the spam filter tolerance, the more false negatives (spam classified as ham). 
For more information see the :ref:`antispam documentation <antispam_howto>`.

Auto-forward
`````````````
Under the section `Auto-forward`, the automatic forwarding of received email messages can be enabled. When enabled, all received email messages are forwarded to the specified email address. 

The option "Keep a copy of the emails" can be ticked, to keep a copy of the received email message in the inbox folder.

In the destination textbox, the email addresses can be entered for automatic forwarding. When entering multiple email addresses the comma (',') must be used as delimiter.

Update password
---------------

On the `update password` page, the password of the logged in user can be changed. Changes are effective immediately. 

.. _webadministration_auto-reply:

Auto-reply
----------

> On the `auto-reply` page, automatic replies can be configured. This is also known as out of office replies.

Reviewer: ```suggestion
On the `auto-reply` page, automatic replies can be configured. This is also known as out of office (ooo) or out of facility (oof) replies.
```

Especially the latter is often used in Microsoft contexts, so I think it makes sense to add them in case someone searches for them with ctrl+f

E.g. if the email subject of the received email message is "how are you?", then the email subject of the automatic reply is `auto: how are you?`.

.. _webadministration_fetched_accounts:

Fetched accounts
----------------

This page is only available when the Fetchmail container is part of your Mailu deployment. 
Fetchmail can be enabled when creating the docker-compose.yml file with the setup utility (https://setup.mailu.io).

On the `fetched accounts` page you can configure email accounts from which email messages will be retrieved.
Only unread email messages are retrieved from the specified email account. 
By default Fetchmail will retrieve email messages every 10 minutes. This can be changed in the Mailu.env file. 
For more information on changing the polling interval see :ref:`the configuration reference <fetchmail>`.

You can add a fetched account by clicking on the `Add an account` button on the top right of the page. To add an fetched account, the following settings must be configured:

* Protocol (IMAP or POP3). The protocol used for accessing the email server.

* Hostname or IP. The hostname or IP address of the email server.

> * TCP port. The TCP port the email server listens on. Common ports are 993 (TLS) or 143 (STARTTLS/unsecure) for IMAP. 

Reviewer: ```suggestion
* TCP port. The TCP port the email server listens on. Common ports are 993 (IMAPS with TLS), 143 (IMAP with STARTTLS), 995 (POP3S with TLS) or 110 (POP3 with STARTTLS).
```

Maybe it's again me reading things wrong, but imo `STARTTLS/unsecure` sounds like STARTTLS itself is unsecure.

Also I don't think plain text/unencrypted is actually supported, as https://github.com/Mailu/Mailu/blob/master/optional/fetchmail/fetchmail.py#L16 specifies `--sslcertck`:

> --sslcertck causes fetchmail to require that SSL/TLS be used and disconnect if it can not successfully negotiate SSL or TLS, or if it cannot successfully verify and validate the certificate and follow it to a trust anchor (or trusted root certificate)

Author: You are correct. At that time I did not know that we do not allow unencrypted connections for pop3 and imap. Good catch.

> * Enable TLS. Tick this setting if the email server requires TLS/SSL.

Reviewer: ```suggestion
* Enable TLS. Tick this setting if the email server requires TLS/SSL instead of STARTTLS.
```

imo this makes it more clearer, that encryption is used in any case.

Reviewer: Also should we use the term SSL? Many people are probably still searching for it, but technically it's deprecated and should just vanish some day. The interface itself also just says TLS, as noted in the beginning of the line.

Author: for now lets stick to SSL. For many people SSL is the same as TLS. We can always remove this in the future once SSL is not commonly used anymore to refer to a secure connection.

> * Username. The user name for logging in the email server. Normally this is the email address or the email address local-part (the part before @). 

Reviewer: ```suggestion
* Username. The user name for logging in to the email server. Normally this is the email address or the email address' local-part (the part before @). 
```

This seems more correct, but I'm definitely not an authority in english.

> * Password. The password for logging in the email server.

Reviewer: ```suggestion
* Password. The password for logging in to the email server.
```

Same as two lines before.

Web administration interface
============================

The web administration interface is the main website for maintaining your Mailu installation. 
For brevity the web administration interface will now be mentioned as admin gui.
It offers the following configuration options:

* change display name.

* change the logged in user's password.

* change user defined spam filter tolerance.

* configure automatic forwarding.

* configure automatic email replies (out of office replies).

> * configure fetchmail for automatic email retrieval.

Author: This makes it more clear. Although the other mail server does not have to be a third party server, I think "(from 3rd-party servers)" is clear enough to convey the message.

.. _webadministration_fetched_accounts:

Fetched accounts
----------------

This page is only available when the Fetchmail container is part of your Mailu deployment. 
Fetchmail can be enabled when creating the docker-compose.yml file with the setup utility (https://setup.mailu.io).

On the `fetched accounts` page you can configure email accounts from which email messages must be retrieved.
Only unread email messages are retrieved from the specified email account. 
By default Fetchmail will retrieve email messages every 10 minutes. This can be changed in the Mailu.env file. 
For more information on changing the polling interval see :ref:`the configuration reference <fetchmail>`.

You can add a fetched account by clicking on the `Add an account` button on the top right of the page. To add an fetched account, the following settings must be configured:

* Protocol (IMAP or POP3). The protocol used for accessing the email server.

* Hostname or IP. The hostname or IP address of the email server.

* TCP port. The TCP port the email server listens on. Common ports are 993 (TLS) or 143 (STARTTLS/unsecure) for IMAP. 

* Enable TLS. Tick this setting if the email server requires TLS/SSL.

> * Username. The user name for logging in the email server. Normally this is the email address or the email address prefix (part before @). 

Author: I'm not certain about this. While "prefix" makes it clear it is before @, it is maybe not how the local part is commonly referred to.
If we would pick a different name, then it should be "local part" instead of "prefix".
[IETF RFC](https://tools.ietf.org/html/rfc3696#section-3) uses "local part' to refer to the part before @.

Maybe change it to "local part"?
What do you think?

> * Username. The user name for logging in the email server. Normally this is the email address or the email address prefix (part before @). 

Author: As agreed we change it to local-part.

* Password. The password for logging in the email server.

* Keep emails on the server. When ticked, retains the email message in the email account after retrieving it.

Click the submit button to apply settings. With the default polling interval, fetchmail will start polling the email account after 10 minutes.

Authentication tokens
---------------------

On the `authentication tokens` page, authentication tokens can be created. Authentications tokens are also known as application passwords.

> The purpose of an authentication token is to create an unique strong password that can be used by a single application to authenticate as the logged in user's email account.

Reviewer: ```suggestion
The purpose of an authentication token is to create a unique and strong password that can be used by a single application to authenticate as the logged in user's email account.
```

When no IP address is entered, there is no restriction on IP address. It is not possible to enter multiple IP addresses.

Announcement
------------

On the `announcement` page, the global administrator can send an email message to all email accounts on the Mailu server. This message will be received as an email message in the inbox folder of each user on the Mailu server.
On the announcement page there are the following options:

* Announcement subject. The subject of the announcement email message.

* Announcement body. The body of the announcement email message.

Click on send to send the announcement email message to all users.

Administrators
--------------

On the `administrators` page, global administrators can be added. A global administrator must be an existing user on the Mailu server.
A global administrator can change `any setting` in the admin GUI. Be careful that you trust the user who you make a global administrator.

Relayed domains
---------------

On the `relayed domains list` page, destination domains can be added that Mailu will relay email messages for without authentication. 

> This means that for these destination domains, other email clients or email servers can send email via Mailu unauthenticated via port 25 to this destination domain. 

Reviewer 2: ```suggestion
This means that for these destination domains, other email clients or email servers can send email via Mailu unauthenticated via port 25 to this destination domain.
Please be extra-careful with this setting, as it could make your mail-server exploitable as a spam-sender in certain scenarios.
```

  As value can be entered either a hostname or IP address of the SMPT server. 
  By default port 25 is used. To use a different port append ":port number" to the Remote Host. For example:
  123.45.67.90:2525.

* Comment. A text field where a comment can be entered to describe the entry.

Changes are effective immediately after clicking the Save button.

NOTE: Due to bug `1588`_ email messages fail to be relayed if no Remote Host is configured. 
As a workaround the HOSTNAME or IP Address of the SMPT server of the relayed domain can be entered as Remote Host.
Please note that no MX lookup is performed when entering a hostname as Remote Host. You can use the MX lookup on mxtoolbox.com to find the hostname and IP Address of the SMTP server.

.. _`1588`: https://github.com/Mailu/Mailu/issues/1588

Antispam
--------

The menu item Antispam opens the Rspamd webgui. For more information how spam filtering works in Mailu see the :ref:`Spam filtering page <antispam_howto_block>`.
The spam filtering page also contains a section that describes how to create a local blacklist for blocking email messages from specific domains. 

> The Rspamd webgui offers basic functions for setting metric actions, scores, viewing statistics and learning.

Reviewer 2: ```suggestion
The Rspamd webgui offers basic functions for setting metric actions, scores, viewing statistics and learning. It is also useful for finding out why certain messages have been accepted or rejected.
```

The following settings are not persisent and are *lost* when the Antispam container is recreated or restarted:

* On the configuration tab, any changes to config files that do not reside in /var/lib or /etc/rspamd/override.d. The last location is mapped to the Mailu overrides folder.

* All information on the History tab.

The following settings are persistent and will survive container recreation:

* All information on the Status tab

* All information on the Throughput tab.

* On the Configuration tab, the changes made to action values (greylist, probably spam ....) and config files that reside in /var/lib or /etc/rspamd/override.d. The last location is mapped to the Mailu overrides folder.

* Any changes made on the Symbols tab.

* Any email messages that have been submitted for spam/ham learning on the Scan/Learn tab.

Mail domains
------------

> On the `Mail domains` page all the domains served by Mailu are configured. Via the new domain button (top right) a new mail domain can be added. For existing domains you can access settings via the icons in the Actions column and Manage column. From left to right you have the following options within the Action column and Manage column.

Reviewer 2: ```suggestion
On the `Mail domains` page all the domains served by Mailu are configured. Via the new domain button (top right) a new mail domain can be added. Please note that you may have to add the new domain to `HOSTNAMES` in your `mailu.env`. For existing domains you can access settings via the icons in the Actions column and Manage column. From left to right you have the following options within the Action column and Manage column.
```

Author: Looks good to me. However because it mentions the mailu.env file. The text mailu.env should actually be a hyperlink to the section HOSTNAMES on the configuration reference page. I will try to create a commit for this when I have time..

Author: Resolved via commit f843bfe.

Details
```````

This page is also accessible for domain managers. On the details page all DNS settings are displayed for configuring your DNS server. It contains information on what to configure as MX record and SPF record. On this page it is also possible to (re-)generate the keys for DKIM and DMARC. The option for generating keys for DKIM and DMARC is only available for global administrators.  After generating the keys for DKIM and DMARC, this page will also show the DNS records for configuring the DKIM/DMARC records on the DNS server. 

Edit
```` 

This page is only accessible for global administrators. On the edit page, the global settings for the domain can be changed.  

* Maximum user count. The maximum amount of users that can be created under this domain. Once this limit is reached it is not possible anymore to add users to the domain; and it is also not possible for users to self-register.
    

> * Maximum alias count. The maximum amount of aliases that can be made for an email account.

Reviewer 2: ```suggestion
* Maximum alias count. The maximum amount of aliases that can be created for an email account.
```

* Edit. For all available options see :ref:`the Add user page <webadministration_add_user>`.

* Delete. Deletes the user. The Admin GUI will ask for confirmation if the user must be really deleted.

* Setting. Access the settings page of the user. See :ref:`the settings page <webadministration_settings>` for more information.

* Auto-reply. Access the auto-reply page of the user. See the :ref:`auto-reply page <webadministration_auto-reply>` for more information.

* Fetched accounts. Access the fetched accounts page of the user. See the :ref:`fetched accounts page <webadministration_fetched_accounts>` for more information.

.. _webadministration_add_user:

Add user
^^^^^^^^

For adding a new user the following options can be configured.

* E-mail. The email address of the new user.

* Password/Confirm password. The password for the new user. The new user can change his password after logging in the Admin GUI.

* Displayed name. The display name of the user within the Admin GUI.

* Comment. A description for the user. This description is shown on the Users page.

> * Enabled. Tick this checkbox to enable the user account. When an user is disabled. The user is unable to login the Admin GUI or webmail or access his email via IMAP/POP3. 

Reviewer 2: ```suggestion
* Enabled. Tick this checkbox to enable the user account. When an user is disabled, the user is unable to login to the Admin GUI or webmail or access his email via IMAP/POP3 or send mail.
```

  The email inbox of the user is still retained. This option can be used to temporarily suspend an user account.
  
* Quota. The maximum quota for the user's email box.

* Allow IMAP access. When ticked, allows email retrieval via the IMAP protocol.

* Allow POP3 access. When ticked, allows email retrieval via the POP3 protocol.

Aliases
```````

This page is also accessible for domain managers. On the aliases page, aliases can be added for email addresses. An alias is a way to disguise another email address. 
Everything sent to an alias email address is actually received in the primary email account's inbox of the destination email address.

> Aliases can diversify a single email account without having to create multiple email addresses (users).

Reviewer 2: ```suggestion
Aliases can diversify a single email account without having to create multiple email addresses (users).
It is also possible to add multiple email addresses to the destination field. All incoming mails will be sent to each users inbox in this case.
```

The following options are available when adding an alias:

* Alias. The alias to create for the specified email address. You cannot use an existing email address.

* Use SQL LIKE Syntax (e.g. for catch-all aliases). When this option is ticked, you can use SQL LIKE syntax as alias.
  The SQL LIKE syntax is used to match text values against a pattern using wildcards. There are two wildcards that can be used with SQL LIKE syntax:
    
    * % - The percent sign represents zero, one, or multiple characters
    * _ - The underscore represents a single character
    
  Examples are:    
    * a% - Finds any values that start with "a"
    * %a - Finds any values that end with "a"
    * %or% - Finds any values that have "or" in any position
    * _r% - Finds any values that have "r" in the second position

>     * a_% -' Finds any values that start with "a" and are at least 2 characters in length

Reviewer 2: ```suggestion
    * a_% - Finds any values that start with "a" and are at least 2 characters in length
```
